driver/maindriver.ml

```diff
@@ -77,6 +77,15 @@ let main argv ppf =
       Compenv.fatal "-set-global-string must be used with -custom, \
                      -output-obj, -output-complete-obj or -output-complete-exe \
                      in bytecode mode.";
+    if !runtime_parameters <> []
+       && (!make_archive
+           || !make_package
+           || (!output_c_object && not !output_complete_executable)) then
+      Compenv.fatal "-set-runtime-default cannot be used with -pack or -a";
+    if !runtime_parameters <> [] && not !custom_runtime then
+      Compenv.fatal "-set-runtime-default must be used with -custom, \
+                     -output-obj, -output-complete-obj or -output-complete-exe \
+                     in bytecode mode.";
```

Comment:
There's not meant to be a limitation on linking mode - the idea here is that _if_ `-set-runtime-default` has been used, there'll be an additional section added to the bytecode image (which is the insight I'd missed when originally writing `-set-global-string`)